### PR TITLE
Plate type field added in 'Dual Index Plate' page

### PR DESCRIPTION
**Pull request by seenanair**
-Mandatory field
-Dropdown list to page, options Fresh frozen and FFPE

### CODE REVIEW

File: src/pages/DualIndexPlate.tsx [hunk 1]
```diff
@@ -163,6 +172,20 @@ function DualIndexPlate() {
               </div>
             </div>
           </div>
+          {sourceReagentPlate && (
+            <div className="w-1/4 mt-4 mb-4" id="plateType">
+              <Label name={'Plate Type'}>
+                <Select emptyOption onChange={(e) => handlePlateTypeChange(e.currentTarget.value)}>
+                  {PLATE_TYPES.map((plateType, index) => (
+                    <option key={index} value={plateType}>
```
Comment (BenTopping) [inline src/pages/DualIndexPlate.tsx:180]: if plateType is unique you may as well use it as the key as well, think index should only be used as last resort key.

File: src/pages/DualIndexPlate.tsx [hunk 2]
```diff
@@ -36,18 +39,24 @@ function DualIndexPlate() {
...
     })
   );
 
-  const { serverErrors, sourceReagentPlate, destLabware, reagentTransfers, workNumber, validationError } =
+  const { serverErrors, sourceReagentPlate, destLabware, reagentTransfers, workNumber, plateType, validationError } =
     current.context;
 
   const handleWorkNumberChange = useCallback(
     (workNumber: string) => {
-      if (workNumber) {
```
Comment (BenTopping) [inline src/pages/DualIndexPlate.tsx:48]: Reason for removing workNumber if check?

File: src/lib/machines/reagentTransfer/reagentTransferMachine.ts [hunk 3]
```diff
@@ -150,6 +162,9 @@ export const reagentTransferMachine = createMachine<ReagentTransferContext, Reag
           UPDATE_TRANSFER_CONTENT: {
             actions: ['assignTransfers']
```
Comment (BenTopping) [inline src/lib/machines/reagentTransfer/reagentTransferMachine.ts:163]: Unrelated to the PR but some of these actions are in lists and others are not, probably good practice to standardise it so all are in lists or none are in lists.

File: src/pages/DualIndexPlate.tsx (hunk 2 above)
Comment (seenanair) [reply src/pages/DualIndexPlate.tsx:48]: If  a worknumber is assigned and then reassign to empty, the validation still pass as we are not updating the worknumber field

File: src/pages/DualIndexPlate.tsx (hunk 1 above)
Comment (seenanair) [reply src/pages/DualIndexPlate.tsx:180]: Yes, Added by accident. plateType is the right field

File: src/lib/machines/reagentTransfer/reagentTransferMachine.ts (hunk 3 above)
Comment (seenanair) [reply src/lib/machines/reagentTransfer/reagentTransferMachine.ts:163]: Done

**Comment by seenanair** [comment]
@BenTopping  Review comments updated.

**Review by BenTopping — APPROVED** [review]
Looks good to me, haven't pulled down and tested as would need to setup core side and data but code looks good.

**Comment by khelwood** [comment]
@seenanair Can you resolve the merge conflicts on this PR please?

**Comment by seenanair** [comment]
> @seenanair Can you resolve the merge conflicts on this PR please?

@khelwood Sure 👍

**Comment by seenanair** [comment]
> @seenanair Can you resolve the merge conflicts on this PR please?

Done

**Pull request merged**